metadata/src/lib.rs

```diff
@@ -631,6 +638,40 @@ pub struct RuntimeApiMethodParamMetadata {
     pub ty: u32,
 }
 
+/// Metadata of custom types with custom values, basically the same as `frame_metadata::v15::CustomMetadata<PortableForm>>`.
+#[derive(Debug, Clone)]
+pub struct CustomMetadata {
+    pub(crate) map: BTreeMap<String, frame_metadata::v15::CustomValueMetadata<PortableForm>>,
+}
+
+impl CustomMetadata {
+    /// Get a certain [CustomMetadataValue] by its name.
+    pub fn get(&self, name: &str) -> Option<CustomMetadataValue<'_>> {
+        self.map.get(name).map(|e| CustomMetadataValue {
+            type_id: e.ty.id,
+            data: &e.value,
+        })
+    }
+}
+
+/// Basically the same as `frame_metadata::v15::CustomValueMetadata<PortableForm>>`, but borrowed.
+pub struct CustomMetadataValue<'a> {
+    type_id: u32,
+    data: &'a [u8],
+}
+
+impl<'a> CustomMetadataValue<'a> {
+    /// the scale encoded value
+    pub fn bytes(&self) -> &[u8] {
+        &self.data
+    }
```

Comment:
```suggestion
    pub fn bytes(&self) -> &'a [u8] {
        &self.data
    }
```

Just to tie the lifetime to that of the bytes and not that of this `CustomMetadataValue` instance

Reply:
Oh yeah true, nice catch!